client.go

```diff
@@ -548,34 +571,52 @@ func (c *Client) Create(cfg Function) (err error) {
 	return
 }
 
+// createRuntimeDir creates a .func directory in the root of the given
+// Function which is also registered as ignored in .gitignore
+func createRuntimeDir(f Function) error {
+	if err := os.MkdirAll(filepath.Join(f.Root, RunDataDir), os.ModePerm); err != nil {
+		return err
+	}
+
+	gitignore := `
+# Functions use the .func directory for local runtime data which should
+# generally not be tracked in source control:
+/.func
+`
+	return os.WriteFile(filepath.Join(f.Root, ".gitignore"), []byte(gitignore), os.ModePerm)
```

Comment:
Thank you yes, this is correct.  I was hoping to jump straight to mutating an existing .gitignore if it exists, but this PR is big enough, so that can be offloaded as an enhancement and for now simply added to contentious files.